src/plugins/region_map/public/components/custom_vector_upload.tsx

```diff
@@ -0,0 +1,290 @@
+  const INDEX_NAME_SPECIAL_CHARACTERS_CHECK = /[`!@#$%^&*()\=\[\]{};':"\\|,.<>\/?~]/;
+  const INDEX_NAME_NOT_BEGINS_WITH_CHECK = /^[-+_.]/;
+  const INDEX_NAME_BEGINS_WITH_CHECK = /^[a-z]/;
+  const INDEX_NAME_NOT_ENDS_WITH_CHECK = /.*-map$/;
+
+  // File size that can be uploaded: 250 MB
+  const MAX_FILE_SIZE_IN_BYTES = 26214400;
+  const MAX_LENGTH_OF_INDEX_NAME = 250;
+
+  const options = [
+    { value: 'geo_shape', text: 'Geo shape' },
+    { value: 'geo_point', text: 'Geo point' },
+  ];
+
+  const [value, setValue] = useState('');
+  const [isLoading, setLoading] = useState(false);
+  const onSelectChange = (e) => {
+    setSelectValue(e.target.value);
+  };
+
+  const [selectValue, setSelectValue] = useState('');
+
+  const onTextChange = (e) => {
+    setValue(e.target.value);
+  };
+
+  const onChange = (files) => {
+    if (files[0]) {
+      validateFileSize(files);
+    }
+  };
+
+  const fetchElementByName = (elementName) => {
+    return document.getElementsByName(elementName)[0];
+  };
+
+  const validateIndexName = (typedIndexName, isIndexNameWithSuffix) => {
+    let error = '';
+    const errorIndexNameDiv = fetchElementByName('errorIndexName');
+
+    // check for presence of index name entered by the user
+    if (!typedIndexName) {
+      error = 'Required';
+    } else {
+      // check for restriction on length of the index name
+      if (MAX_LENGTH_OF_INDEX_NAME < typedIndexName.length) {
+        error += ' Map name should be less than 250 characters.';
+      }
+
+      // check for restriction on the usage of upper case characters in the index name
+      if (INDEX_NAME_UPPERCASE_CHECK.test(typedIndexName)) {
+        error += ' Upper case letters are not allowed.';
+      }
+
+      // check for restriction on the usage of special characters in the index name
+      if (INDEX_NAME_SPECIAL_CHARACTERS_CHECK.test(typedIndexName)) {
+        error += ' Special characters are not allowed.';
+      }
+
+      // check for restriction on the usage of characters at the beginning in the index name
+      if (
+        INDEX_NAME_NOT_BEGINS_WITH_CHECK.test(typedIndexName) ||
+        !INDEX_NAME_BEGINS_WITH_CHECK.test(typedIndexName)
+      ) {
+        error += " Map name can't start with + , _ , - or . It should start with a-z.";
+      }
+
+      // check for restriction on the usage of -map in map name if entered by the user
+      if (!isIndexNameWithSuffix && INDEX_NAME_NOT_ENDS_WITH_CHECK.test(typedIndexName)) {
+        error += " Map name can't end with -map";
+      }
+    }
+
+    if (error) {
+      errorIndexNameDiv.textContent = error;
+      return false;
+    } else {
+      errorIndexNameDiv.textContent = '';
+      return true;
+    }
+  };
+
+  const validateFileSize = async (files) => {
+    // check if the file size is permitted
+    if (MAX_FILE_SIZE_IN_BYTES < files[0].size) {
+      notifications.toasts.addWarning('File size should be less than 25 MB.');
```

Comment:
Constant for limit. Hence, if we change it future, we don't have to update here

Reply:
Adding a constant. Will change it to error as well.